### PR TITLE
iOS 16 context menu

### PR DESCRIPTION
**Pull request by justinmc**
This PR implements the iOS 16 light mode context menu visuals.  It does not change the animation, which is different in iOS 16 and will need to be updated in a separate PR.

Partial fix for: https://github.com/flutter/flutter/issues/111671

| Platform | Native iOS 16 light mode | Native iOS 16 dark mode |
| --- | --- | --- |
| Flutter (this branch) | Screenshot 2022-11-21 at 4 50 36 PM | Screenshot 2022-11-21 at 4 44 41 PM |
| Native iOS 16 | image | image |

 
https://user-images.githubusercontent.com/389558/203186678-086f04a1-040f-4efc-aade-b9de1be15161.mov

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/text_selection_toolbar.dart [hunk 1]
```diff
@@ -31,7 +32,20 @@ const Radius _kToolbarBorderRadius = Radius.circular(8);
 
 // Colors extracted from https://developer.apple.com/design/resources/.
 // TODO(LongCatIsLooong): https://github.com/flutter/flutter/issues/41507.
-const Color _kToolbarDividerColor = Color(0xFF808080);
+const Color _kToolbarDividerColorDark = Color(0xFF808080);
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/cupertino/text_selection_toolbar.dart:35]: use CupertinoDynamicColor instead?

Comment (justinmc) [reply packages/flutter/lib/src/cupertino/text_selection_toolbar.dart]: Yes thank you, I forgot that existed but it's so much more convenient 👍

**Review by MitchellGoodwin — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/cupertino/text_selection_toolbar.dart [hunk 2]
```diff
@@ -119,14 +138,24 @@ class CupertinoTextSelectionToolbar extends StatelessWidget {
   // Builds a toolbar just like the default iOS toolbar, with the right color
   // background and a rounded cutout with an arrow.
   static Widget _defaultToolbarBuilder(BuildContext context, Offset anchor, bool isAbove, Widget child) {
-    return _CupertinoTextSelectionToolbarShape(
+    final bool isDarkMode = MediaQuery.of(context).platformBrightness == Brightness.dark;
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/cupertino/text_selection_toolbar.dart:141]: Use `CupertinoTheme.brightnessOf` instead?

File: packages/flutter/lib/src/cupertino/text_selection_toolbar_button.dart [hunk 3]
```diff
@@ -104,17 +109,28 @@ class CupertinoTextSelectionToolbarButton extends StatelessWidget {
...
+             ? textColor
+             : CupertinoColors.inactiveGray,
+       ),
+     );
+
     return CupertinoButton(
       borderRadius: null,
-      color: _kToolbarBackgroundColor,
-      disabledColor: _kToolbarBackgroundColor,
+      color: backgroundColor,
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/cupertino/text_selection_toolbar_button.dart:132]: same comment as above. I think CupertinoButton resolves dynamic colors so you can just construct a dyanmic color and give it to the `CupertinoButton` here

**Review by LongCatIsLooong — APPROVED** [review]

**Pull request merged**